src/serac/numerics/functional/tensor.hpp

```diff
@@ -939,6 +939,25 @@ SERAC_HOST_DEVICE constexpr auto operator*(const tensor<S, m...>& A, const tenso
   return dot(A, B);
 }
 
+/**
+ * @brief cross product for dim 3 vectors
+ * @tparam S the underlying type of the first (lefthand) vector argument
+ * @tparam T the underlying type of the second (righthand) vector argument
+ * @tparam m first dimension of A and B (must be 3)
+ * @param[in] A The lefthand vector
+ * @param[in] B The righthand vector
+ */
+template <typename S, typename T, int m>
+SERAC_HOST_DEVICE constexpr auto cross(const tensor<S, m>& A, const tensor<T, m>& B)
+{
+  static_assert(m == 3, "Cross products are only defined on vectors of dimension 3.");
+
+  tensor<decltype(S{} * T{}), m> cross_product{
+      {A[1] * B[2] - A[2] * B[1], A[2] * B[0] - A[0] * B[2], A[0] * B[1] - A[1] * B[0]}};
+
+  return cross_product;
+}
```

Comment:
Whoops, I don't know how we made it this far without a cross product!

If you want to enforce that dim == 3 (which we should), you can just put that directly in the function signature, and then you don't need the static assert,

```suggestion
template <typename S, typename T>
SERAC_HOST_DEVICE constexpr  tensor<decltype(S{} * T{}), 3> cross(const tensor<S, 3>& u, const tensor<T, 3>& v)
{
  return {u[1] * v[2] - u[2] * v[1], u[2] * v[0] - u[0] * v[2], u[0] * v[1] - u[1] * v[0]}};
}
```

It's also often useful to define a couple special cases of "2D" cross product:

```cpp
template <typename S, typename T>
SERAC_HOST_DEVICE constexpr  auto cross(const tensor<S, 2>& u, const tensor<T, 2>& v) {
  return u[0] * v[1] - u[1] * v[0]; // out-of-plane component, really just det({u, v})
}

template <typename T>
SERAC_HOST_DEVICE constexpr tensor<T, 2> cross(const tensor<T, 2>& u) {
  return {-u[1], u[0]}; // in-plane result of cross(vec3{0, 0, 1}, vec3{u[0], u[1], 0})
}
```

I'm writing this code in the browser, so it might have some mistakes